itests/util/src/main/java/org/apache/hadoop/hive/ql/QTestMetaStoreHandler.java

```diff
@@ -98,14 +98,12 @@ private String getDbTypeConfString() {// "ORACLE", "MYSQL", "MSSQL", "POSTGRES"
   }
 
   public void beforeTest() throws Exception {
-    getRule().before();
-    if (!isDerby()) {// derby is handled with old QTestUtil logic (TxnDbUtil stuff)
-      getRule().install();
+    if (isDerby()) {
```

Comment:
If I understood it correctly, `DatabaseRule.before` initialises the docker instance. So that change says docker will initialised once for the whole test suite, not once per test case.
And we have to keep it for Derby as it sets the test directory (`test.tmp.dir`). 

Am I correct?

Reply:
Yes, since if the Postgre would be reinitialized ,then the tests would fail as it is mentioned in the ticket ("Database does not exist: default")